swupd/manifest.go

```diff
@@ -319,6 +319,32 @@ func (m *Manifest) WriteManifest(w io.Writer) error {
 	return nil
 }
 
+// filterFilesVersion returns a new manifest with only Files and DeletedFiles
+// updated in versions greater than fromVersion
+func (m *Manifest) filterFilesVersion(fromVersion uint32) *Manifest {
+	fm := &Manifest{
+		Header:     m.Header,
+		Name:       m.Name,
+		BundleInfo: m.BundleInfo,
+	}
+	fm.Header.ContentSize = 0
+
+	for _, f := range m.Files {
+		if f.Version > fromVersion {
+			fm.AppendFile(f)
+		}
+	}
+
+	for _, f := range m.DeletedFiles {
```

Comment:
Is this second for loop redundant? It looks like m.DeletedFiles is a subset of m.Files, so the deleted files are iterated over twice.

Reply:
oh, I didn't noticed that. Good catch.